### PR TITLE
Mutations

### PR DESCRIPTION
**Pull request by CorneliaKelinske**
Do you like my mutaaaaaations?

And the tests for them?
Quick review please! :) thank you!

### CODE REVIEW

File: lib/graphql_practice_web/schema/mutations/upload.ex [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  alias GraphqlPracticeWeb.Resolvers
+
+  object :upload_mutations do
+    @desc "Creates an upload"
+    field :create_upload, :upload_result do
+      arg :title, non_null(:string)
+      arg :description, non_null(:string)
+      arg :user_id, non_null(:id)
+
+      resolve(&Resolvers.Upload.create_upload/3)
```
Comment (alanvardy) [inline lib/graphql_practice_web/schema/mutations/upload.ex]:
```suggestion
      resolve &Resolvers.Upload.create_upload/3
```

File: lib/graphql_practice_web/schema/mutations/user.ex [hunk 2]
```diff
@@ -0,0 +1,13 @@
+defmodule GraphqlPracticeWeb.Schema.Mutations.User do
+  use Absinthe.Schema.Notation
+  alias GraphqlPracticeWeb.Resolvers
+
+  object :user_mutations do
+    @desc "Creates a user"
+    field :create_user, :user_result do
+      arg :name, non_null(:string)
+      arg :email, non_null(:string)
+      resolve(&Resolvers.User.create_user/3)
```
Comment (alanvardy) [inline lib/graphql_practice_web/schema/mutations/user.ex]:
```suggestion
      resolve &Resolvers.User.create_user/3
```

File: test/graphql_practice_web/schema/mutations/upload_test.exs [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+                   "description" => create_description,
+                   "user_id" => user_id
+                 }
+               )
+
+      assert %{"createUpload" => %{"errors" => nil, "upload" => upload}} = data
+
+      assert Map.get(upload, "title") === create_title
+      assert Map.get(upload, "description") === create_description
+      assert get_in(upload, ["user", "id"]) === to_string(user_id)
```
Comment (alanvardy) [inline test/graphql_practice_web/schema/mutations/upload_test.exs]: Pattern matching might be nicer for this

File: test/graphql_practice_web/schema/mutations/upload_test.exs [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+      string_id = Map.get(upload, "id")
+      id = String.to_integer(string_id)
+
+      assert %Upload{title: ^create_title, description: ^create_description, id: ^id} =
+               Content.get_upload!(id)
```
Comment (alanvardy) [inline test/graphql_practice_web/schema/mutations/upload_test.exs:60]: Nice

File: test/graphql_practice_web/schema/mutations/upload_test.exs [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+      assert {:ok, %{errors: errors}} =
+               Absinthe.run(@create_upload_doc, Schema,
+                 variables: %{
+                   "title" => create_title,
+                   "description" => create_description,
+                   "user_id" => user_id
+                 }
+               )
+
+      assert [%{details: _details, message: _message, path: _path}] = errors
```
Comment (alanvardy) [inline test/graphql_practice_web/schema/mutations/upload_test.exs]: What is this testing?

Maybe you can `assert message =~ "something"`

**Review by alanvardy — APPROVED** [review]
Looks good, left some suggestions

File: test/graphql_practice_web/schema/mutations/upload_test.exs (hunk 3 above)
Comment (CorneliaKelinske) [reply test/graphql_practice_web/schema/mutations/upload_test.exs]: Changed it. Thank you!

File: test/graphql_practice_web/schema/mutations/upload_test.exs (hunk 5 above)
Comment (CorneliaKelinske) [reply test/graphql_practice_web/schema/mutations/upload_test.exs]: thank you! I fixed it!

**Pull request merged**